internal/venice-common/src/main/java/com/linkedin/venice/kafka/consumer/ApacheKafkaConsumer.java

```diff
@@ -152,6 +178,15 @@ public ConsumerRecords<byte[], byte[]> poll(long timeoutMs) {
     while (attemptCount <= consumerPollRetryTimes && !Thread.currentThread().isInterrupted()) {
       try {
         records = kafkaConsumer.poll(Duration.ofMillis(timeoutMs));
+        polledPubSubMessages.clear();
+        for (TopicPartition topicPartition: records.partitions()) {
+          PubSubTopicPartition pubSubTopicPartition = assignments.get(topicPartition);
+          for (ConsumerRecord<byte[], byte[]> consumerRecord: records.records(topicPartition)) {
+            PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> pubSubMessage =
+                pubSubMessageDeserializer.deserialize(consumerRecord, pubSubTopicPartition);
+            polledPubSubMessages.computeIfAbsent(pubSubTopicPartition, k -> new ArrayList<>()).add(pubSubMessage);
+          }
```

Comment:
Can we fine tune this code a bit? I suggested we:

1. Put the list coming out of `records.records(topicPartition)` in a local variable,
2. Create our own list by passing the correct size parameter in it (same as the size of the list that came out of Kafka), 
3. Populate our list with deserialized objects,
4. Then put the list in the map to be returned (i.e. no need perform a lookup per message on that map, when one per partition will suffice).